edgelet/iotedged/src/lib.rs

```diff
@@ -184,6 +191,11 @@ impl Main {
             }
         }
 
+        if let Provisioning::External(ref external) = settings.provisioning() {
+            // Set the hosting endpoint environment variable for use by the custom HSM library.
```

Comment:
I'm not a fan of using one-off env vars to communicate with specific HSM implementations. It would be nice if there was a strongly-typed way of passing this through instead.

I talked with @mrohera and he said there isn't such a way today. What would be needed to do it?

Reply:
Easiest way I can think of is having structs in the hsm_client_data.h file that have these environment variables as members and the callers can initialize these structs with the right values and pass them along to the hsm library when calling the init method of each interface.

The env vars can be grouped together into different structs based on semantics.

That being said, I can make that change outside of this pull-request as that'll require other changes that don't directly relate to what this PRs objective is.

Reply:
I think we should create a section of customizable key value pairs in the config.yaml. ESD would then package these and send them down to HSM layer. This does require updates to the HSM layer by adding a new API and thus is a bigger task.

Reply:
I think it is fine since we have a prior precedent and not able to offer a cleaner solution at this time. The endpoint needs to be known to the Edgelet as well as the HSM library and so it is in the config.yaml. The other way was the HSM lib would use a config file private to the HSM but this involves duplication of the end point configuration data.

Reply:
Yep, that's why leaving it as is for now. We can fix this later in a different change to make it strongly typed